bqn.html.markdown

```diff
@@ -0,0 +1,289 @@
+---
+language: bqn
+filename: learnbqn.bqn
+contributors:
+    - ["Raghu Ranganathan", "https://github.com/razetime"]
+translators:
+---
+
+BQN is a modern array language (similar to APL) that aims to eliminate burdensome aspects of the APL tradition.
+
+It is recommended to try these code examples out in a REPL. The [online REPL](https://mlochbaum.github.io/BQN/try.html) is
+recommended for quick start, since it comes with keyboard and easy to access help. You can try building
+[CBQN](https://github.com/dzaima/CBQN) for a local install, but it will need keyboard setup.
+
+```bqn
+# This is a comment.
+# The characters ',' and `⋄` are statement separators.
+
+##################
+# Main datatypes #
+##################
+
+# Numbers
+1,2,3,4
+¯1,¯2,¯3  # Negative numbers are written with a high minus
+π,∞,¯π,¯∞ # Pi and Infinity are defined constants
+1_234_456 # You can add underscores in between numbers
+          # This does not change their value
+1.3E4     # Scientific notation is supported
+
+# Characters
+'a','⥊'
+'
+'         # Yes, you can put *any* character in a character literal
+@         # Null character ('\0' in C)
+# Arrays
+1‿2‿3       # Stranding, good for simple lists
+⟨1,2,3⟩     # General list notation
+⟨1‿2,2‿3⟩   # Both can be mixed
+[1‿2,2‿3]   # Array notation
+            # An array is multidimensional, as opposed to containing sublists.
+            # It must be rectangular in shape (a grid structure rather than a tree structure)
+[1‿2‿3,4‿5] # This is hence invalid
+            # May be familiar coming from Numpy, MATLAB and similar languages.
+"asdf"      # Character array (String)
+"newline
+separated"  # Allows newlines
+"quo""tes"  # Escape a double quote by typing it twice
+# Functions
+1{𝕨+𝕩}3       # All functions are infix
```

Comment:
That looks wrong, I need to go through this later

Reply:
may i ask what exactly is wrong?

Reply:
Of course, BQN's parsing is sane, but I think this might be confusing to people coming from context-sensitive APLs: (this is Dyalog APL)
```apl
      {⍵}4{⍵}3
3
      {⍵}4{⍺⋄⍵}3
4
```
It might not be obvious that BQN is actually sane, and thus such cases do not happen, but my initial "wrong" assessment is wrong